File: Day12/perfectNumber.js
 // Function for check if number is perfect number or not
function isPerfectNumber(num){
    if(num <= 1){
        return 0;
    }
    let sum =0;
    for(let i =1; i<=(num/2); i++){
        if(num%i==0){
            sum += i;
        }
    }
    return sum==num;
}
//Let's start here
let num =parseInt(prompt("Enter the number:"))

//Call the function and display result
let ans = isPerfectNumber(num);

if(ans)
    console.log(num + " is a perfect number");
else
    console.log(num + "is not a perfect number");

File: Day12/perfect_number.py
""" program to check number is perfect number or not"""

def is_perfect_number(num):
    """Function for check if number is perfect number or not"""
    if num <= 1:
        return 0
    total_sum=0
    for i in range(1,(num+2)//2,1):
        if num%i==0:
            total_sum += i
    return total_sum==num

#Let's start here
number = int(input("Enter the number: "))
#Call the function and display result
ans = is_perfect_number(number)
if ans:
    print(number," is a perfect number.")
else:
    print(number," is not a perfect number.")
